### PR TITLE
Readme file

### PR DESCRIPTION
**Pull request by Climbatize**
# Description
Updated readme file

## Links
MINI-869

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,17 +34,55 @@ Generated documentation for this SDK can be found at https://rakutentech.github.
 
 To run the example project, clone the repo, and run `pod install` from the Example directory first.
 
-## Requirements
+### Custom parameters
+
+In order to load your MiniApp, you will have to use your own Host App ID and your subscription key. These can eiher be set in project configuration plist (RASApplicationIdentifier and RASProjectSubscriptionKey) or by taping the top right configuration icon in the example application.
```
Comment (corycaywood) [inline README.md:39]: Need to also mention that they should set the base url. On android, we also have a message like this: `We don't currently host a public API, so you will need to provide your own Base URL for API requests.`.

File: README.md [hunk 2]
```diff
@@ -16,17 +34,55 @@ Generated documentation for this SDK can be found at https://rakutentech.github.
...
+MiniApp is available through [CocoaPods](https://cocoapods.org). To install it, simply add the following line to your Podfile:
 
 ```ruby
 pod 'MiniApp'
 ```
 
+In your project configuration .plist you can add 2 custom string target properties :
+
+- `RASApplicationIdentifier` - to set your MiniApp host application identifier
+- `RASProjectSubscriptionKey` - to set your MiniApp subsbscription key
```
Comment (corycaywood) [inline README.md:52]: Also should list the base url property here.

File: README.md [hunk 3]
```diff
@@ -6,7 +6,25 @@
 
 # MiniApp
 
-This SDK provides capabilities to downloading/cache and host the Mini-App in native iOS applications. It also provides more features to interact with the native application via bridge.
+This open-source library allows you to integrate a Rakuten MiniApp into your iOS applications. It also provides more features to interact with the native application via bridge.
+
+## Features
+
+- Load MiniApp list
+- Load MiniApp infos
```
Comment (ts-leojoseph-r) [inline README.md:14]: Maybe information or meta-information ?

File: README.md [hunk 4]
```diff
@@ -6,7 +6,25 @@
...
+
+## Getting started
+
+### Requirements
+
+This module supports iOS 11.0 and above. It has been tested on iOS 11.0 and above.
+
+It is written in Swift 5.0 and can be used in compatible Xcode versions.
+
+In order to run your MiniApp you will have to provide your MiniApp host application identifier and subsbscription key to the library
```
Comment (ts-leojoseph-r) [inline README.md:27]: Subscription key?

File: README.md [hunk 5]
```diff
@@ -16,17 +34,55 @@ Generated documentation for this SDK can be found at https://rakutentech.github.
...
+MiniApp.shared().info(miniAppId: miniAppID) { (result) in
+	...
+}
+```
+
+Create a MiniApp for the given `MiniAppInfo` object :
+
+```
+MiniApp.shared().create(appInfo: info, completionHandler: { (result) in
+	...
```
Comment (corycaywood) [inline README.md:82]: I think the example for `create` should be a little more complete.  On Android side, there is mostly complete Activity example showing how they would set the mini app to the view after retrieving it.

File: README.md (hunk 4 above)
Comment (ts-leojoseph-r) [reply README.md:27]: I guess Endpoint detail is missing here

File: README.md (hunk 1 above)
Comment (ts-leojoseph-r) [reply README.md:39]: Agree with cory + eiher -> either

File: README.md [hunk 6]
```diff
@@ -16,17 +34,55 @@ Generated documentation for this SDK can be found at https://rakutentech.github.
...
+
+If you don't want to use project settings, you are still able to pass this informations one by one to the `Config.userDefaults` using a `Config.Key` as key:
+
+```
+Config.userDefaults?.set("MY_CUSTOM_ID", forKey: Config.Key.subscriptionKey.rawValue)
+```
+
+## Usage
+
+MiniApp library calls are done via the `MiniApp.shared()` singleton with or without a `MiniAppSdkConfig` instance (you can get the current one with `Config.getCurrent()`). If you don't provide a config instance, values in custiom iOS target properties will be used by default. For example, if you want to load the MiniApp list:
```
Comment (ts-leojoseph-r) [inline README.md:62]: custiom -> custom?

File: README.md [hunk 7]
```diff
@@ -16,17 +34,55 @@ Generated documentation for this SDK can be found at https://rakutentech.github.
...
+
+```
+MiniApp.shared(with: Config.getCurrent()).list { (result) in
```
Comment (ts-leojoseph-r) [inline README.md:65]: Do you think its a good idea to add another approach as well i.e w/o passing the Config

File: README.md (hunk 6 above)
Comment (Climbatize) [reply README.md:62]: 😅

File: README.md (hunk 1 above)
Comment (Climbatize) [reply README.md:39]: You're right, I base the doc on the custom runtime parameters we had in the application

File: README.md (hunk 7 above)
Comment (Climbatize) [reply README.md:65]: yes it is 👍

File: README.md (hunk 2 above)
Comment (Climbatize) [reply README.md:52]: done

**Comment by ts-leojoseph-r** [comment]
> Oh, sorry I just noticed something else. Can you also check the public interface docs?
> 
> I noticed that `RealMiniAppView` is public but it should be private. Most of the public interface has documentation, but please check that all of it does. I think `MiniAppInfo` is missing documentation.

Thanks for spotting it @corycaywood. Done now

**Pull request merged**